pkg/image/registryclient/client_mirrored.go

```diff
@@ -197,7 +215,18 @@ func (r *blobMirroredRepository) attemptFirstConnectedRepo(ctx context.Context,
 			}
 			continue
 		}
-		return fn(repo)
+		if err := fn(repo); err != nil {
+			if !isRequestError(err) {
```

Comment:
Why in those particular cases we would return instead of falling back to other registries?